modules/xmpp/JingleSessionPC.js

```diff
@@ -2030,6 +2031,49 @@ export default class JingleSessionPC extends JingleSession {
             });
     }
 
+    /**
+     * Adds a new track to the peerconnection. This method needs to be called only when a secondary JitsiLocalTrack is
+     * being added to the peerconnection for the first time.
+     *
+     * @param {JitsiLocalTrack} localTrack track to be added to the peer connection.
+     * @returns {Promise<void>} that resolves when the track is successfully added to the peerconnection, rejected
+     * otherwise.
+     */
+    addTrack(localTrack) {
+        if (!this.usesUnifiedPlan || localTrack.type !== MediaType.VIDEO) {
+            return Promise.reject(new Error('Multiple tracks of a given media type are not supported'));
+        }
+
+        const workFunction = finishedCallback => {
+            const remoteSdp = new SDP(this.peerconnection.peerconnection.remoteDescription.sdp);
+
+            // Add a new transceiver by adding a new mline in the remote description.
```

Comment:
Interesting, so the mlines are added by injecting into remote sdp. Will `addTransceiver` not work here?

Reply:
Even with addTransceiver, the remote description would still have to be munged before setting the remote description after the transceiver is added. I played around with addTransceiver a bit but it resulted in the browser renegotiation loop triggered by 'negotiationneeded' event fired by the browser.